django_jsonform/contrib/dataclasses/models/fields.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING, Any, List, Optional, Type
+
+from django_jsonform.contrib.dataclasses.forms.fields import DataclassJSONFormField
+from django_jsonform.contrib.dataclasses.typedefs import DataclassJsonSchema, SerializableValue
+from django_jsonform.contrib.dataclasses.utils import json_schema_array
+from django_jsonform.exceptions import JSONSchemaValidationError
+from django_jsonform.models.fields import JSONField
+
+if TYPE_CHECKING:
+    from django.db import models
+    from django.db.backends.base.base import BaseDatabaseWrapper
+    from django.db.models.expressions import Expression
+
+
+class DataclassJSONField(JSONField):
+    form_class = DataclassJSONFormField
+
+    def __init__(
+        self,
+        dataclass_cls: Type[DataclassJsonSchema],
+        *,
+        many: bool = False,
```

Comment:
This `many` convention comes from Django Rest Framework

But we can consider making this a `DataclassArrayJSONField` instead